### DISCUSSION TITLE
How to run PTB with asyncio in a thread (together with a Flask web server)

### DISCUSSION THREAD

**Discussion by ferraridavide** [python-telegram-bot/python-telegram-bot#3516] [Q&A]
I would like to begin by stating that I am not an expert in asyncio, but I hope that sharing my experience can be beneficial for others as well, particularly since the transition to asyncio has recently been made. 
I am currently working on building a bot where users can subscribe to be notified when certain "events" occur, the script receives these requests from the bot and subscribes to a webhook for the event. Once the webhook calls back, the user receives a message.

To listen for the webhook, I am using Flask, which as far as I know, does not utilize asyncio. 
The `app.run(...)` function in Flask is blocking, but I can run it in a separate thread without any issues. 
The `application.run_pooling()` function in PTB is also blocking and when I attempt to run it on a separate thread, I encounter an error: `RuntimeError: There is no current event loop in thread.` I have tried various methods such as `asyncio.run_coroutine_threadsafe` and `asyncio.create_task,` but I have been unable to successfully implement them.

I am wondering if the correct approach would be to put both Flask and PTB on two new threads and if so, how can I successfully run PTB on a thread? If not, what alternative solution should I explore?

Thank you for your help.

**Comment by Bibo-Joshi** [comment]
You have already been pointed to the relevant wiki section in the user group: https://t.me/pythontelegrambotgroup/652977?thread=652971 😉

(As I see no mention on `input()` in your problem description this time, I assume that it's not relevant anymore)

**Reply by lpares12** [reply]
This didn't work for me. The title says "Running PTB alongside other asyncio frameworks", and Flask is not an asyncio framework. But anyways I thought it would be useful so I tried.

Even with a small example, I noticed that the function `post_init` is never called. Even if I add a `while(True)` after `await application.updater.start_polling()`.

Is this expected?

EDIT: Also, I send commands and it does not seem to receive them.

**Reply by Bibo-Joshi** [reply]
Please be sure to check out the "customwebhookbot" examples. We have provided a flask version of that example.
